audio/load.go

```diff
@@ -49,70 +50,128 @@ func Get(filename string) (Data, error) {
 // one stored in the loeaded map.
 func Load(directory, filename string) (Data, error) {
 	dlog.Verb("Loading", directory, filename)
-	if !IsLoaded(filename) {
-		f, err := fileutil.Open(filepath.Join(directory, filename))
-		if err != nil {
-			return nil, err
-		}
-		defer f.Close()
-		var buffer audio.Audio
-		end := strings.ToLower(filename[len(filename)-4:])
-		switch end {
-		case ".wav":
-			buffer, err = wav.Load(f)
-		case ".mp3":
-			buffer, err = mp3.Load(f)
-		default:
-			return nil, oakerr.UnsupportedFormat{Format: end}
-		}
-		if err != nil {
-			return nil, err
-		}
-		loaded[filename] = buffer.(audio.FullAudio)
+	if data, ok := getLoaded(filename); ok {
+		return data, nil
 	}
-	return loaded[filename], nil
+	f, err := fileutil.Open(filepath.Join(directory, filename))
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	var buffer audio.Audio
+	end := strings.ToLower(filename[len(filename)-4:])
+	switch end {
+	case ".wav":
+		buffer, err = wav.Load(f)
+	case ".mp3":
+		buffer, err = mp3.Load(f)
+	default:
+		return nil, oakerr.UnsupportedFormat{Format: end}
+	}
+	if err != nil {
+		return nil, err
+	}
+	data := buffer.(audio.FullAudio)
+	setLoaded(filename, data)
+	return data, nil
 }
 
 // Unload removes an element from the loaded map. If the element does not
 // exist, it does nothing.
 func Unload(filename string) {
+	loadedLock.Lock()
 	delete(loaded, filename)
+	loadedLock.Unlock()
 }
 
 // IsLoaded is shorthand for (if _, ok := loaded[filename]; ok)
```

Comment:
Given the lock usage perhaps this comment should be updated?

Reply:
Discussed offline-- Don't think the lock itself is significant enough to change the comment, the presence of the lock around the map should be transparent to the user